### PR TITLE
Spice merchant

### PR DESCRIPTION
**Pull request by gopz**

### CODE REVIEW

File: sets/hinterlands.py [hunk 1]
```diff
@@ -168,6 +168,41 @@ def get_vp(self):
 		victory_cards = [x for x in cards if "Victory" in x.type]
 		return int(len(victory_cards) / 4)
 
+class Spice_Merchant(crd.Card):
+	def __init__(self, game, played_by):
+		crd.Card.__init__(self, game, played_by)
+		self.title = 'Spice Merchant'
+		self.description = 'You may trash a treasure from your hand.'\
+		 "If you do {}, {} or {}, {}".format(crd.format_draw(2), crd.format_actions(1), crd.format_money(2), crd.format_buys(1))
```
Comment (hphu) [inline sets/hinterlands.py]: please make the +1 action +2 cards and +$2 +1 buy options on the same line (total of 2 lines) using the second inline format parameter `crd.format_actions(1, True)` with a new line after the `If you do`

File: sets/hinterlands.py [hunk 2]
```diff
@@ -168,6 +168,41 @@ def get_vp(self):
...
+		treasure_titles = list(set(map(lambda x: x.title, treasure_cards)))
+		if len(treasure_titles) == 0:
+			self.game.announce("-- but there were no treasures in hand")
+			crd.Card.on_finished(self, False, False)
+		else:	
+			card_selection = yield self.played_by.select(None, 1, treasure_titles, "Choose a treasure card to trash")
+			if card_selection:
+				self.played_by.discard(card_selection, self.game.trash_pile)
+				perk_selection = yield self.played_by.select(1, 1, ["+2 cards, +1 action", "+2$, +1 buy"], 
+					"Choose one: +2 Cards, +1 Action; or +2$, +1 Buy")
```
Comment (hphu) [inline sets/hinterlands.py]: remove commas, make +2$ -> +$2

File: sets/hinterlands.py [hunk 3]
```diff
@@ -168,6 +168,41 @@ def get_vp(self):
...
+		crd.Card.play(self, skip)
+		treasure_cards = self.played_by.hand.get_cards_by_type("Treasure", True)
+		treasure_titles = list(set(map(lambda x: x.title, treasure_cards)))
+		if len(treasure_titles) == 0:
+			self.game.announce("-- but there were no treasures in hand")
+			crd.Card.on_finished(self, False, False)
+		else:	
+			card_selection = yield self.played_by.select(None, 1, treasure_titles, "Choose a treasure card to trash")
+			if card_selection:
+				self.played_by.discard(card_selection, self.game.trash_pile)
```
Comment (hphu) [inline sets/hinterlands.py]: Need to announce trashed card to game `self.game.announce(...)`

**Comment by gopz** [comment]
@refbot I'm not sure what your talking about, please make line comments so I can tell what you're referring to

**Comment by gopz** [comment]
Please merge if everything looks good

**Pull request merged**